Fix printing of errors in juliac + searchsorted functions for LinearAlgebra

gbaraldi opened the pull request:
We may want to move this changes to base proper. 
@LilithHafner should sort functions just always specialize on the `by`/`lt` params? It currently sometimes relies on inlining for good performance

gbaraldi committed:
Fix printing of errors in juliac + searchsorted functions for LinearAlgebra

gbaraldi committed:
Also disable logging

topolarity commented on contrib/juliac/juliac-trim-base.jl:
```diff
@@ -109,3 +118,7 @@ end
 end
+
+@eval Base.CoreLogging begin
+    @inline current_logger_for_env(std_level::LogLevel, group, _module) = nothing
```
```suggestion
    # Disable logging (TypedCallable is required to support the existing dynamic
    # logger interface, but it's not implemented yet)
    @inline current_logger_for_env(std_level::LogLevel, group, _module) = nothing
```

topolarity commented on contrib/juliac/juliac-trim-base.jl:
```diff
@@ -82,6 +82,15 @@ end
+# Used for LinearAlgebre ldiv with SVD
+for s in [:searchsortedfirst, :searchsortedlast, :searchsorted]
+    @eval Base.Sort begin
+        $s(v::AbstractVector, x, o::Ordering) = $s(v,x,firstindex(v),lastindex(v),o)
```
```suggestion
        # identical to existing Base def. but specializes on `lt` / `by`
        $s(v::AbstractVector, x, o::Ordering) = $s(v,x,firstindex(v),lastindex(v),o)
```

topolarity committed:
Increase `throw_eachindex_mismatch_indices` specialization

Allows `eachindex(A::Vector{Float64}, B::Vector{Float64})` to be
`--trim`-compatible.

gbaraldi committed:
Apply suggestions from code review

Co-authored-by: Cody Tapscott <84105208+topolarity@users.noreply.github.com>

LilithHafner commented:
Sounds good to include in Base. @gbaraldi, would you share an example of one of the `searchsorted` functions having poor performance when not inlined? (feel free to add `@noinline` statements if necessary)

gbaraldi commented:
It's not even poor performance. It's just not typestable if not inlined
